pypeit/spectrographs/gtc_osiris.py

```diff
@@ -0,0 +1,337 @@
+
+        Returns:
+            `numpy.ndarray`_: Boolean array with the flags selecting the
+            exposures in ``fitstbl`` that are ``ftype`` type frames.
+        """
+        good_exp = framematch.check_frame_exptime(fitstbl['exptime'], exprng)
+        if ftype in ['science','standard']:
+            return good_exp & (fitstbl['target'] != 'ArcLamp_Xe') \
+            & (fitstbl['target'] != 'ArcLamp_HgAr') \
+            & (fitstbl['target'] != 'ArcLamp_Ne') \
+            & (fitstbl['target'] != 'SpectralFlat') \
+                    & (fitstbl['target'] != 'BIAS')
+        if ftype in ['arc', 'tilt']:
+            return good_exp & ((fitstbl['target'] == 'ArcLamp_Xe') \
+            | (fitstbl['target'] == 'ArcLamp_HgAr') \
+            | (fitstbl['target'] == 'ArcLamp_Ne'))
+        if ftype in ['pixelflat', 'trace', 'illumflat']:
+            return good_exp & (fitstbl['target'] == 'SpectralFlat')
+        if ftype == 'bias':
+            return good_exp & (fitstbl['target'] == 'BIAS')
+
+        msgs.warn('Cannot determine if frames are of type {0}.'.format(ftype))
+        return np.zeros(len(fitstbl), dtype=bool)
+
+    def config_independent_frames(self):
+        """
+        Define frame types that are independent of the fully defined
+        instrument configuration.
+
+        Bias and dark frames are considered independent of a configuration.
+        Standards are assigned to the correct configuration frame group by
+        grism (i.e. ignoring that they are taken with a wider slit).
+        See :func:`~pypeit.metadata.PypeItMetaData.set_configurations`.
+
+        Returns:
+            :obj:`dict`: Dictionary where the keys are the frame types that
+            are configuration independent and the values are the metadata
+            keywords that can be used to assign the frames to a configuration
+            group.
+        """
+        return {'standard': 'dispname','bias': None, 'dark': None}
+
+
+    def config_specific_par(self, scifile, inp_par=None):
+        """
+        Modify the ``PypeIt`` parameters to hard-wired values used for
+        specific instrument configurations.
+
+        Args:
+            scifile (:obj:`str`):
+                File to use when determining the configuration and how
+                to adjust the input parameters.
+            inp_par (:class:`~pypeit.par.parset.ParSet`, optional):
+                Parameter set used for the full run of PypeIt.  If None,
+                use :func:`default_pypeit_par`.
+
+        Returns:
+            :class:`~pypeit.par.parset.ParSet`: The PypeIt parameter set
+            adjusted for configuration specific parameter values.
+        """
+        # Start with instrument wide
+        par = super().config_specific_par(scifile, inp_par=inp_par)
+
+        # Wavelength calibrations
+        if self.get_meta_value(scifile, 'dispname') == 'R300B':
+            par['calibrations']['wavelengths']['wv_cen'] = 4405.
```

Comment:
For the record, `wv_cen` and `disp` aren't used.
But no harm including them.

Reply:
Ok, this was left over from when I was trying to use holy-grail (which fails for most of these gratings).